src/os.zig

```diff
@@ -0,0 +1,281 @@
+//! OS-level functionality.
+//! Designed to be slightly higher level than sys.zig
+
+/// Opaque type representing the user ID.
+pub const Uid = struct {
+    const Self = @This();
+
+    _underlying: std.c.uid_t,
+
+    pub fn queryEffective() Self {
+        // TODO(markovejnovic): Unfortunately, geteuid is not available in Zig's stdlib at the
+        //                      moment, so we need to use @cImport above.
+        return .{ ._underlying = bun.c.geteuid() };
+    }
+
+    pub fn queryReal() Self {
+        // TODO(markovejnovic): Unfortunately, getuid is not available in Zig's stdlib at the
+        //                      moment, so we need to use @cImport above.
+        return .{ ._underlying = bun.c.getuid() };
+    }
+
+    /// Falls back to queryEffective.
+    pub fn queryCurrent() Self {
+        return queryEffective();
+    }
+};
+
+/// Utilities for fetching and interacting with the current system home directory.
+pub const HomeDir = struct {
+    pub const QueryOpts = struct {
+        user: ?[]const u8 = null,
+    };
+
+    const Self = @This();
+
+    path: pathlib.AutoAbsPath,
```

Comment:
this should be `OSPathSlice`. `AutoAbsPath` uses an object pool of allocated path buffers, but we don't need that for this situation